xline/src/server/watch_server.rs

```diff
@@ -111,7 +113,9 @@ where
                 _ = ticker.tick() => {
                     watch_handle.handle_tick_progress().await;
                 }
-                _ = stop_notify.listen() => {
+                // To ensure that each iteration invokes the same `stop_listener` and keeps
+                // events losing due to the cancellation of `stop_listener` at bay.
+                _ = &mut stop_listener => {
```

Comment:
Why it is cancellation safe to use the same `stop_listener`

Reply:
In fact, `stop_listener` is not cancellation safe. Due to the ownership of `stop_listner` is not belong to `select!`, it won't be canceled by `select!`.